python/pyarrow/hdfs.py

```diff
@@ -146,8 +146,12 @@ def _derive_hadoop_classpath():
     xargs_echo = subprocess.Popen(('xargs', 'echo'),
                                   stdin=find.stdout,
                                   stdout=subprocess.PIPE)
-    return subprocess.check_output(('tr', "' '", "':'"),
-                                   stdin=xargs_echo.stdout)
+    classpath_jars = subprocess.check_output(('tr', "' '", "':'"),
+                                             stdin=xargs_echo.stdout)
+    classpath_conf = os.environ["HADOOP_CONF_DIR"] \
+        if "HADOOP_CONF_DIR" in os.environ \
```

Comment:
could you comment on what you are trying to do (pointers to documentation woudl be helpful.  given the number of breakages with this code is possible to add a unit test?

Reply:
This PR  modify the _derive_hadoop_classpath method: Add hadoop configuration directory to hadoop classpath.
This methos will first check $HADOOP_CONF_DIR env ,if HADOOP_CONF_DIR is set ,add $HADOOP_CONF_DIR value to hadoop classpath,if HADOOP_CONF_DIR is not set ,add $HADOOP_HOME/etc/haoop to hadoop classpath